lib/Page.js

```diff
@@ -62,10 +63,55 @@ function calculateNewBaseUrl(filePath, root, lookUp) {
   return calculate(filePath, []);
 }
 
+function insertFlexDivBeforeLastFooter(pageData) {
+  const $ = cheerio.load(pageData);
+  const footers = $('footer');
+  if (footers.length > 0) {
+    const lastFooter = footers.last();
+    if (lastFooter.before().attr('id') !== 'flexDiv') {
+      $(lastFooter).before(FLEX_DIV_HTML);
+    }
+  }
+
+  return new Promise((resolve) => {
+    resolve($.html());
+  });
+}
+
 function unique(array) {
   return array.filter((item, pos, self) => self.indexOf(item) === pos);
 }
 
+function removeDuplicateFooters(pageData) {
+  const $ = cheerio.load(pageData);
+  const footers = $('footer');
+  const flexDivs = $('#flexDiv');
+  if (footers.length > 1) {
+    const lastFooter = footers.last();
+    footers.remove();
+    flexDivs.last().after(lastFooter);
+  }
+
+  return new Promise((resolve) => {
+    resolve($.html());
+  });
+}
+
+function removeFooterParentTags(pageData) {
+  const $ = cheerio.load(pageData);
+  const lastFooter = $('footer').last();
+  const lastFooterParentNum = lastFooter.parents().length;
```

Comment:
Does `lastFooter.parent()` give you the nested HTML elements? If so, is it possible to just go to the highest layer and replace that?

Reply:
Iterative method changed to constant time method:
```javascript
  if (lastFooterParentNum > 0) {
    const lastParentOfFooter = lastFooter.parents().last();
    const footerParentContents = lastFooter.parent().contents();
    lastParentOfFooter.replaceWith(footerParentContents);
  }
```